SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/LayeredLBStrategy.cpp

```diff
@@ -0,0 +1,274 @@
+	{
+		return false;
+	}
+
+	return LayerNameToLBStrategy[LayerName]->ShouldHaveAuthority(Actor);
+}
+
+VirtualWorkerId ULayeredLBStrategy::WhoShouldHaveAuthority(const AActor& Actor) const
+{
+	if (!IsReady())
+	{
+		UE_LOG(LogLayeredLBStrategy, Warning, TEXT("LayeredLBStrategy not ready to decide on authority for Actor %s."), *AActor::GetDebugName(&Actor));
+		return SpatialConstants::INVALID_VIRTUAL_WORKER_ID;
+	}
+
+	const FName& LayerName = GetLayerForActor(Actor);
+	if (!LayerNameToLBStrategy.Contains(LayerName))
+	{
+		UE_LOG(LogLayeredLBStrategy, Warning, TEXT("LayeredLBStrategy doesn't have a LBStrategy for Actor %s which is in Layer %s."), *AActor::GetDebugName(&Actor), *LayerName.ToString());
+		return SpatialConstants::INVALID_VIRTUAL_WORKER_ID;
+	}
+
+	const VirtualWorkerId ReturnedWorkerId = LayerNameToLBStrategy[LayerName]->WhoShouldHaveAuthority(Actor);
+
+	UE_LOG(LogLayeredLBStrategy, Log, TEXT("LayeredLBStrategy returning virtual worker id %d for Actor %s."), ReturnedWorkerId, *AActor::GetDebugName(&Actor));
+	return ReturnedWorkerId;
+}
+
+SpatialGDK::QueryConstraint ULayeredLBStrategy::GetWorkerInterestQueryConstraint() const
+{
+	check(IsReady());
+	if (!VirtualWorkerIdToLayerName.Contains(LocalVirtualWorkerId))
+	{
+		UE_LOG(LogLayeredLBStrategy, Warning, TEXT("LayeredLBStrategy doesn't have a LBStrategy for worker %d."), LocalVirtualWorkerId);
+		SpatialGDK::QueryConstraint Constraint;
+		Constraint.ComponentConstraint = 0;
+		return Constraint;
+	}
+	else
+	{
+		return LayerNameToLBStrategy[VirtualWorkerIdToLayerName[LocalVirtualWorkerId]]->GetWorkerInterestQueryConstraint();
+	}
+}
+
+FVector ULayeredLBStrategy::GetWorkerEntityPosition() const
+{
+	check(IsReady());
+	if (!VirtualWorkerIdToLayerName.Contains(LocalVirtualWorkerId))
+	{
+		return FVector{ 0.f, 0.f, 0.f };
+	}
+	else
+	{
+		return LayerNameToLBStrategy[VirtualWorkerIdToLayerName[LocalVirtualWorkerId]]->GetWorkerEntityPosition();
+	}
+}
+
+uint32 ULayeredLBStrategy::GetMinimumRequiredWorkers() const
+{
+	// The MinimumRequiredWorkers for this strategy is a sum of the required workers for each of the wrapped strategies.
+	uint8 MinimumRequiredWorkers = 0;
+	for (const auto& Elem : LayerNameToLBStrategy)
+	{
+		MinimumRequiredWorkers += Elem.Value->GetMinimumRequiredWorkers();
+	}
+
+	UE_LOG(LogLayeredLBStrategy, Verbose, TEXT("LayeredLBStrategy needs %d workers to support all layer strategies."), MinimumRequiredWorkers);
+	return MinimumRequiredWorkers;
+}
+
+void ULayeredLBStrategy::SetVirtualWorkerIds(const VirtualWorkerId& FirstVirtualWorkerId, const VirtualWorkerId& LastVirtualWorkerId)
+{
+	// If the LayeredLBStrategy wraps { SingletonStrategy, 2x2 grid, Singleton } and is given IDs 1 through 6 it will assign:
+	// Singleton : 1
+	// Grid : 2 - 5
+	// Singleton: 6
+	VirtualWorkerId NextWorkerIdToAssign = FirstVirtualWorkerId;
+	for (auto& Elem : LayerNameToLBStrategy)
+	{
+		VirtualWorkerId MinimumRequiredWorkers = Elem.Value->GetMinimumRequiredWorkers();
```

Comment:
Big Nit: Might be a little more readable if you pull out into local variables:
FName LayorName = Elem.Key;
LBS LoadbalancingStrategy = Elem.Value;

Up to you...

Reply:
I made a local for the Strategy, but not the LayerName, because it's only used one place, where it's pretty obvious the type.